File: app/src/main/java/com/example/collections_and_maps/RecyclerSizeLookup.java
package com.example.collections_and_maps;


import androidx.recyclerview.widget.GridLayoutManager;

public class RecyclerSizeLookup extends GridLayoutManager.SpanSizeLookup {

    private final int spanPos, spanCnt1, spanCnt2;

    public RecyclerSizeLookup(int spanPos, int spanCnt1, int spanCnt2) {
        super();
        this.spanPos = spanPos;
        this.spanCnt1 = spanCnt1;
        this.spanCnt2 = spanCnt2;
    }

    @Override
    public int getSpanSize(int position) {
        return (position % spanPos == 0 ? spanCnt2 : spanCnt1);
    }
}

File: app/src/main/java/com/example/collections_and_maps/ui/benchmark/RecyclerSizeLookup.kt
package com.example.collections_and_maps.ui.benchmark

import androidx.recyclerview.widget.GridLayoutManager.SpanSizeLookup

class RecyclerSizeLookup(
    private val spanPos: Int,
    private val spanCnt1: Int,
    private val spanCnt2: Int
) : SpanSizeLookup() {
    override fun getSpanSize(position: Int): Int {
        return if (position % spanPos == spanCnt2) spanCnt2 else spanCnt1
    }
}